GenumKit/Stencil/Contexts.swift

```diff
@@ -80,13 +83,17 @@ extension StoryboardParser {
       if let scenes = storyboardsScenes[storyboardName] {
         sbMap["scenes"] = scenes
           .sort({$0.storyboardID < $1.storyboardID})
-          .map { (scene: Scene) -> [String:String] in
-            // Handle special scene.tag cases like navigationController, splitViewController, etc…
-            // TODO: Fix this to extract the 'UI' prefix and be able to implement OSX support in #128
-            let customClass = scene.customClass ??
-              (scene.tag != "viewController" ? "UI" + uppercaseFirst(scene.tag) : nil)
-            if let customClass = customClass {
-              return ["identifier": scene.storyboardID, "class": customClass]
+          .map { (scene: Scene) -> [String:AnyObject] in
+            if scene.customClass == nil && scene.tag == "viewController" {
+              return [
+                "identifier": scene.storyboardID,
+                "baseType": uppercaseFirst(scene.tag),
+                "isBaseViewController": true
+              ]
+            } else if scene.customClass == nil {
+                return ["identifier": scene.storyboardID, "baseType": uppercaseFirst(scene.tag)]
+            } else if let customClass = scene.customClass {
+              return ["identifier": scene.storyboardID, "customClass": customClass]
             } else {
               return ["identifier": scene.storyboardID]
```

Comment:
We can't do that (or I missed  feature from `stencil`). In order to check if it's a `isBaseViewController`, we are using `{% ifnot scene.isBaseViewController %}` which consist into simply check if the key exists are no whatever is the value.

Anyway, you are still right that the current conditions are more complex than they should do.